cmd/README.md

```diff
@@ -0,0 +1,738 @@
+	}
+	// no input provided for this function
+
+	// CUSTOM CODE STARTS HERE
+	// Get the current state
+	currentState := accessibleState.GetStateDB()
+	// Get the value set at recipient
+	value := currentState.GetState(HelloWorldAddress, common.BytesToHash([]byte("storageKey")))
+	// Do some processing and pack the output
+	packedOutput, err := PackSayHelloOutput(string(common.TrimLeftZeroes(value.Bytes())))
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+```
+
+Finally we can modify our `setGreeting()` function. 
+
+``` go
+func setGreeting(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SetGreetingGasCost); err != nil {
+		return nil, 0, err
+	}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+	// attempts to unpack [input] into the arguments to the SetGreetingInput.
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+    inputStr, err := UnpackSetGreetingInput(input)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// CUSTOM CODE STARTS HERE
+    // check if the input string is longer than 32 bytes
+    if len(inputStr) > 32 {
+      return nil, 0, errors.New("input string is longer than 32 bytes")
+    }
+
+	// setGreeting is the execution function "SetGreeting(name string)"
+    // and sets the storageKey in the string returned by hello world
+  
+    res := common.LeftPadBytes([]byte(inputStr), common.HashLength)
+	accessibleState.GetStateDB().SetState(HelloWorldAddress, common. BytesToHash([]byte("storageKey")), common.BytesToHash(res))
+
+	// this function does not return an output, leave this one as is
+	packedOutput := []byte{}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+```
+
+## Step 4: Add Upgradable Config
+
+Let's now modify `params/precompile_config.go`. We can `CTRL F` for `ADD YOUR PRECOMPILE HERE`. 
+
+Let's add our key. Note that there are two places to in the image below. 
+
+![](2022-09-01-23-20-53.png)
+
+Let's add our precompile config to `PrecompileUpgrade`
+
+![](2022-09-01-23-19-42.png)
+
+Finally, we can add a getter function. 
+
+![](2022-09-01-23-23-20.png)
+
+Done! All we had to do was follow the comments.
+
+
+## Step 5: Add Precompile Upgrade
+
+Let's add our precompile upgrade in `params/config.go`. We can `CTRL F` for `ADD YOUR PRECOMPILE HERE`. 
+
+Let's add the bool to check if our precompile is enabled.  
+
+![](2022-09-01-23-35-47.png)
+
+We can now add it to the Avalanche rules. 
+
+![](2022-09-01-23-36-56.png)
+
+Lastly, we can add `IsHelloWorld` which checks if we are equal or greater than the fork `blockTimestamp`. 
```

Comment:
We should change the order of steps here. The `IsHelloWorld` function is used by the previous step, so we should first add this function, then add it to rules, and then set it in the `AvalancheRules(...)` function.

Reply:
thanks